ssl/statem/statem_lib.c

```diff
@@ -704,6 +704,19 @@ static int version_cmp(const SSL *s, int a, int b)
     return DTLS_VERSION_LT(a, b) ? -1 : 1;
 }
 
+int ssl_strict_version_check(const SSL *s, int version)
```

Comment:
I think this still isn't quite right. Compare this with the checks in `ssl_method_error`.